cmd/rekor-cli/app/pflags.go

```diff
@@ -130,14 +131,79 @@ func validateArtifactPFlags(uuidValid, indexValid bool) error {
 		if signature == "" && typeStr == "rekord" {
 			return errors.New("--signature is required when --artifact is used")
 		}
-		if publicKey == "" {
+		if publicKey == "" && typeStr != "jar" {
 			return errors.New("--public-key is required when --artifact is used")
 		}
 	}
 
 	return nil
 }
 
+func CreateJarFromPFlags() (models.ProposedEntry, error) {
+	//TODO: how to select version of item to create
+	returnVal := models.Jar{}
+	re := new(jar_v001.V001Entry)
+
+	jar := viper.GetString("entry")
+	if jar != "" {
+		var jarBytes []byte
+		jarURL, err := url.Parse(jar)
+		if err == nil && jarURL.IsAbs() {
+			/* #nosec G107 */
+			rpmResp, err := http.Get(jar)
+			if err != nil {
+				return nil, fmt.Errorf("error fetching 'jar': %w", err)
+			}
+			defer rpmResp.Body.Close()
+			jarBytes, err = ioutil.ReadAll(rpmResp.Body)
+			if err != nil {
+				return nil, fmt.Errorf("error fetching 'jar': %w", err)
+			}
+		} else {
+			jarBytes, err = ioutil.ReadFile(filepath.Clean(jar))
+			if err != nil {
+				return nil, fmt.Errorf("error processing 'jar' file: %w", err)
+			}
+		}
+		if err := json.Unmarshal(jarBytes, &returnVal); err != nil {
+			return nil, fmt.Errorf("error parsing jar file: %w", err)
+		}
+	} else {
+		// we will need only the artifact; public-key & signature are embedded in JAR
+		re.JARModel = models.JarV001Schema{}
+		re.JARModel.Archive = &models.JarV001SchemaArchive{}
+
+		artifact := viper.GetString("artifact")
+		dataURL, err := url.Parse(artifact)
+		if err == nil && dataURL.IsAbs() {
+			re.JARModel.Archive.URL = strfmt.URI(artifact)
+		} else {
+			artifactBytes, err := ioutil.ReadFile(filepath.Clean(artifact))
+			if err != nil {
+				return nil, fmt.Errorf("error reading artifact file: %w", err)
+			}
+
+			//TODO: ensure this is a valid JAR file; look for META-INF/MANIFEST.MF?
+			re.JARModel.Archive.Content = strfmt.Base64(artifactBytes)
+		}
+
+		if err := re.Validate(); err != nil {
+			return nil, err
+		}
+
+		if re.HasExternalEntities() {
```

Comment:
Would a jar ever have external entities? I think this is always false?

Reply:
This is meant to declare whether the JSON doc posted to Rekor needs to fetch an external entity (i.e. download the JAR file from an HTTP(S) server).

Reply:
Ahhh right, the entire jar can be external.